src/saveload/saveload.cpp

```diff
@@ -2906,19 +2906,19 @@ static SaveOrLoadResult DoLoad(std::shared_ptr<LoadFilter> reader, bool load_che
 	if (_sl.lf->Read((uint8_t*)hdr, sizeof(hdr)) != sizeof(hdr)) SlError(STR_GAME_SAVELOAD_ERROR_FILE_NOT_READABLE);
 
 	/* see if we have any loader for this type. */
-	const SaveLoadFormat *fmt = _saveload_formats;
+	const SaveLoadFormat *fmt = std::begin(_saveload_formats);
 	for (;;) {
 		/* No loader found, treat as version 0 and use LZO format */
-		if (fmt == endof(_saveload_formats)) {
+		if (fmt == std::end(_saveload_formats)) {
```

Comment:
this loop could do with a bit of reorganisation, imo

```
for (_saveload_formats) {
   ...
}

if (not loaded) {
   assert(find(lzo_support));
   try_lzo;
}
```

?

Reply:
It could indeed be improved, but that would be a massive refactor as it'd be better to move all the logic for choosing the `SaveLoadFormat` into a separate function. That way you can return the format when you found it, instead of extra checks. That would make it better organised, but well beyond the scope for this PR.